src/gui/editors/PianoRoll.cpp

```diff
@@ -2811,8 +2811,8 @@ void PianoRoll::dragNotes( int x, int y, bool alt, bool shift, bool ctrl )
 				if (note->selected())
 				{
 					int newLength = note->oldLength() + off_ticks;
-					newLength = qMax(1, newLength);
-					note->setLength( MidiTime(newLength) );
+					newLength = qMax(alt ? 1 : quantization(), newLength);
```

Comment:
Implementation for my preferred behavior, since code might be easier to interpret than my explanation 😅 
```suggestion
					if (newLength <= 0) { newLength = quantization(); }
```
Or if you prefer
```suggestion
					newLength = (newLength > 0) ? newLength : quantization();
```

Reply:
That's elegant! I prefer the <Alt> key in there somewhere. You can skip quantization when changing the length of the note with the <Alt> key so I would prefer:

```cpp
if (newLength <= 0){ newLength = alt ? 1 : quantization(); }
```

If you change that I'll commit it.

Reply:
Like so? :)
```suggestion
					if (newLength <= 0) { newLength = alt ? 1 : quantization(); }
```

Reply:
Yup!

Reply:
Reminder that `newLength = alt` is not a boolean operation, but an assignment one. 😉

Reply:
Yes. We're assigning either the quantized value or if overriding with Alt key, 1. Then you can still set the note to any length if you want.

Reply:
Okay! Wanted to make sure since tertiary operator can be confusing sometimes.